integration-tests/tests/src/tests/sync/user.ts

```diff
@@ -574,4 +574,20 @@ describe.skipIf(environment.missingServer, "User", () => {
       await expect(user.functions.error()).to.be.rejectedWith("function not found: 'error'");
     });
   });
+  describe("currentUser", () => {
+    importAppBefore(buildAppConfig("with-custom-function").anonAuth());
+
+    it("persists currentUser on opening the app", async function (this: AppContext & RealmContext) {
+      const credentials = Realm.Credentials.anonymous();
+      const user = await this.app.logIn(credentials);
+      const appId = this.app.id;
+      //@ts-expect-error Normally you wouldn't do this
+      delete this.app;
+      this.app = new Realm.App(appId);
+
+      const currentUser = this.app.currentUser;
+      expect(currentUser).not.to.be.null;
+      expect(user.id).to.equal(currentUser?.id);
```

Comment:
Wouldn't it be simpler to just have a new local app instance here?

```suggestion
      const newApp = new Realm.App(appId);

      const currentUser = newApp.currentUser;
      expect(currentUser).not.to.be.null;
      expect(user.id).to.equal(currentUser?.id);
```